From 78d475b31e634a42a041b9cd01563761ad984a08 Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <jan.steffens@gmail.com>
Date: Mon, 30 Mar 2015 08:59:51 +0200
Subject: [PATCH] Mark p11-kit's trust stores as built-in

Needed for the crypto glue integration via replacing libnssckbi.so.
---
 security/certverifier/CertVerifier.cpp | 8 ++++++++
 toolkit/modules/CertUtils.jsm          | 4 +++-
 2 files changed, 11 insertions(+), 1 deletion(-)

diff --git a/security/certverifier/CertVerifier.cpp b/security/certverifier/CertVerifier.cpp
index fca5f89..81d3ad2 100644
--- a/security/certverifier/CertVerifier.cpp
+++ b/security/certverifier/CertVerifier.cpp
@@ -77,6 +77,14 @@ IsCertBuiltInRoot(CERTCertificate* cert, bool& result) {
       result = true;
       return SECSuccess;
     }
+    if (strcmp("Default Trust", token) == 0) {
+      result = true;
+      return SECSuccess;
+    }
+    if (strcmp("System Trust", token) == 0) {
+      result = true;
+      return SECSuccess;
+    }
   }
   return SECSuccess;
 }
diff --git a/toolkit/modules/CertUtils.jsm b/toolkit/modules/CertUtils.jsm
index 00a2c52..309412e 100644
--- a/toolkit/modules/CertUtils.jsm
+++ b/toolkit/modules/CertUtils.jsm
@@ -174,7 +174,9 @@ this.checkCert =
 }
 
 function isBuiltinToken(tokenName) {
-  return tokenName == "Builtin Object Token";
+  return tokenName == "Builtin Object Token" ||
+         tokenName == "Default Trust" ||
+         tokenName == "System Trust";
 }
 
 /**
-- 
2.3.4

